From 4e6bbfdca5cb144de47150db8ba786da737fb23d Mon Sep 17 00:00:00 2001
From: Kay Sievers <kay@vrfy.org>
Date: Fri, 26 Oct 2012 03:43:56 +0200
Subject: [PATCH 211/214] udev: builtin - do not fail builtin initialization if
 one of them returns an error (cherry picked from commit
 4af113f9971217991cce51b45f15bdce22ee45e2)

Conflicts:
	src/udev/udev-builtin-hwdb.c
---
 src/udev/udev-builtin.c | 24 ++++++++----------------
 src/udev/udev.h         |  2 +-
 2 files changed, 9 insertions(+), 17 deletions(-)

diff --git a/src/udev/udev-builtin.c b/src/udev/udev-builtin.c
index b632eda..8de3c60 100644
--- a/src/udev/udev-builtin.c
+++ b/src/udev/udev-builtin.c
@@ -42,24 +42,18 @@ static const struct udev_builtin *builtins[] = {
 #endif
 };
 
-int udev_builtin_init(struct udev *udev)
+void udev_builtin_init(struct udev *udev)
 {
         unsigned int i;
-        int err = 0;
 
         if (initialized)
-                return 0;
+                return;
 
-        for (i = 0; i < ELEMENTSOF(builtins); i++) {
-                if (builtins[i]->init) {
-                        err = builtins[i]->init(udev);
-                        if (err < 0)
-                                break;
-                }
-        }
+        for (i = 0; i < ELEMENTSOF(builtins); i++)
+                if (builtins[i]->init)
+                        builtins[i]->init(udev);
 
         initialized = true;
-        return err;
 }
 
 void udev_builtin_exit(struct udev *udev)
@@ -79,13 +73,11 @@ void udev_builtin_exit(struct udev *udev)
 bool udev_builtin_validate(struct udev *udev)
 {
         unsigned int i;
-        bool change = false;
 
         for (i = 0; i < ELEMENTSOF(builtins); i++)
-                if (builtins[i]->validate)
-                        if (builtins[i]->validate(udev))
-                                change = true;
-        return change;
+                if (builtins[i]->validate && builtins[i]->validate(udev))
+                        return true;
+        return false;
 }
 
 void udev_builtin_list(struct udev *udev)
diff --git a/src/udev/udev.h b/src/udev/udev.h
index c1fe8d1..b8b3bd8 100644
--- a/src/udev/udev.h
+++ b/src/udev/udev.h
@@ -167,7 +167,7 @@ extern const struct udev_builtin udev_builtin_pci_db;
 extern const struct udev_builtin udev_builtin_usb_db;
 extern const struct udev_builtin udev_builtin_usb_id;
 extern const struct udev_builtin udev_builtin_uaccess;
-int udev_builtin_init(struct udev *udev);
+void udev_builtin_init(struct udev *udev);
 void udev_builtin_exit(struct udev *udev);
 enum udev_builtin_cmd udev_builtin_lookup(const char *command);
 const char *udev_builtin_name(enum udev_builtin_cmd cmd);
-- 
1.8.1.5

